include/directx/d3dx12.h

```diff
@@ -4037,6 +4038,277 @@ class CD3DX12_NODE_MASK_SUBOBJECT
+    FEATURE_SUPPORT_GET(D3D12_TILED_RESOURCES_TIER, m_dOptions, TiledResourcesTier);
+    FEATURE_SUPPORT_GET(D3D12_RESOURCE_BINDING_TIER, m_dOptions, ResourceBindingTier);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, PSSpecifiedStencilRefSupported);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, TypedUAVLoadAdditionalFormats);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, ROVsSupported);
+    FEATURE_SUPPORT_GET(D3D12_CONSERVATIVE_RASTERIZATION_TIER, m_dOptions, ConservativeRasterizationTier);
+    FEATURE_SUPPORT_GET(UINT, m_dOptions, MaxGPUVirtualAddressBitsPerResource);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, StandardSwizzle64KBSupported);
+    FEATURE_SUPPORT_GET(D3D12_CROSS_NODE_SHARING_TIER, m_dOptions, CrossNodeSharingTier);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, CrossAdapterRowMajorTextureSupported);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, VPAndRTArrayIndexFromAnyShaderFeedingRasterizerSupportedWithoutGSEmulation);
+    FEATURE_SUPPORT_GET(D3D12_RESOURCE_HEAP_TIER, m_dOptions, ResourceHeapTier);
+
+    // 1: Architecture (Deprecated)
+    // FEATURE_SUPPORT_GET_NODE_INDEXED(BOOL, m_dArchitecture, TileBasedRenderer);
+    // FEATURE_SUPPORT_GET_NODE_INDEXED(BOOL, m_dArchitecture, UMA);
+    // FEATURE_SUPPORT_GET_NODE_INDEXED(BOOL, m_dArchitecture, CacheCoherentUMA);
+
+    // 2: Feature Levels
+    // User inputs a list of feature levels. The function returns the highest level supported.
+    // Option 1: Straightforward implementation. Only perform forwarding
+    HRESULT HighestFeatureLevel(UINT NumFeatureLevels, const D3D_FEATURE_LEVEL* pFeatureLevelRequested, D3D_FEATURE_LEVEL& MaxSupportedFeatureLevel)
+    {
+        D3D12_FEATURE_DATA_FEATURE_LEVELS dFeatureLevels;
+        dFeatureLevels.NumFeatureLevels = NumFeatureLevels;
+        dFeatureLevels.pFeatureLevelsRequested = pFeatureLevelRequested;
+        
+        HRESULT result = m_pDevice->CheckFeatureSupport(D3D12_FEATURE_FEATURE_LEVELS, &dFeatureLevels, sizeof(D3D12_FEATURE_DATA_FEATURE_LEVELS));
+
+        MaxSupportedFeatureLevel = dFeatureLevels.MaxSupportedFeatureLevel;
+        return result;
+    }
+
+    // Option 2: Checks if a specific feature level is supported. Returns S_OK if supported and DXGI_ERROR_UNSUPPORTED otherwise.
+    HRESULT FeatureLevelSupported(D3D_FEATURE_LEVEL FeatureLevelRequested)
+    {
+        D3D_FEATURE_LEVEL aFLevels[1];
+        aFLevels[0] = FeatureLevelRequested;
+        D3D12_FEATURE_DATA_FEATURE_LEVELS dFeatureLevels;
+        dFeatureLevels.NumFeatureLevels = 1;
+        dFeatureLevels.pFeatureLevelsRequested = aFLevels;
+
+        HRESULT result = m_pDevice->CheckFeatureSupport(D3D12_FEATURE_FEATURE_LEVELS, &dFeatureLevels, sizeof(D3D12_FEATURE_DATA_FEATURE_LEVELS));
+
+        return result;
+    }
+
+    // 3: Feature Format Support
+    HRESULT FormatSupport(DXGI_FORMAT Format, D3D12_FORMAT_SUPPORT1& Support1, D3D12_FORMAT_SUPPORT2& Support2)

```

Comment:
I kept the same call routine for the more complicated feature checks except that I replaced the input struct with flat function parameters. I kept the return type HRESULT since the check may return non-trivial errors (not caused by invalid parameters but due to certain format not supported). Probably not the best way to do so?